src/agent/src/rpc.rs

```diff
@@ -1906,6 +1917,91 @@ mod tests {
         assert!(result.is_err(), "expected add arp neighbors to fail");
     }
 
+    #[tokio::test]
+    async fn test_get_memory_info() {
+        #[derive(Debug)]
+        struct TestData<'a> {
+            // if None is provided, no files will be generated, else the data in the Option will populate the file
+            block_size_data: Option<&'a str>,
+            hotplug_probe_data: Option<&'a str>,
+
+            result: (u64, bool),
+            error_contains: &'a str,
+        }
+
+        let tests = &[
+            TestData {
+                block_size_data: Some("10000000"),
+                hotplug_probe_data: Some("hotplug data"),
+                result: (268435456, true),
+                error_contains: "",
+            },
+            TestData {
+                block_size_data: Some("100"),
+                hotplug_probe_data: None,
+                result: (256, false),
+                error_contains: "",
+            },
+            TestData {
+                block_size_data: None,
+                hotplug_probe_data: None,
+                result: (0, false),
+                error_contains: "",
+            },
+            TestData {
+                block_size_data: Some(""),
+                hotplug_probe_data: None,
+                result: (0, false),
+                error_contains: "Invalid block size",
+            },
+            TestData {
+                block_size_data: Some("-1"),
+                hotplug_probe_data: None,
+                result: (0, false),
+                error_contains: "Invalid block size",
```

Comment:
Looking at `get_memory_info()`, I see that this string is hard-coded twice, so I'd consider creating a `const` for it. You could then check for:

```rust
const ERR_INVALID_BLOCK_SIZE: &str = "Invalid block size";
```

Your test could then check for:

```rust
result: Err(anyhow!(ERR_INVALID_BLOCK_SIZE)),
```